cmsis/arm_math.h

```diff
@@ -293,14 +293,30 @@
 #ifndef _ARM_MATH_H
 #define _ARM_MATH_H
 
-/* ignore some GCC warnings */
-#if defined ( __GNUC__ )
+/* Compiler specific diagnostic adjustment */
+#if   defined ( __CC_ARM )
+
+#elif defined ( __ARMCC_VERSION ) && ( __ARMCC_VERSION >= 6010050 )
+
+#elif defined ( __GNUC__ )
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wsign-conversion"
 #pragma GCC diagnostic ignored "-Wconversion"
 #pragma GCC diagnostic ignored "-Wunused-parameter"
+
+#elif defined ( __ICCARM__ )
+
+#elif defined ( __TI_ARM__ )
+
+#elif defined ( __CSMC__ )
```

Comment:
Ah it's part of the one squashed commits you prepared. It's weird because I got that file from you, to fix CLZ intrinsic on IAR, and I havent modified it. @0xc0170 did you do the modifications?